builtin/logical/database/path_config_connection.go

```diff
@@ -344,16 +344,14 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 
 		b.Logger().Debug("created database object", "name", name, "plugin_name", config.PluginName)
 
-		b.Lock()
-		defer b.Unlock()
-
 		// Close and remove the old connection
-		b.clearConnection(name)
-
-		b.connections[name] = &dbPluginInstance{
+		oldConn := b.connPut(name, &dbPluginInstance{
 			database: dbw,
 			name:     name,
 			id:       id,
+		})
+		if oldConn != nil {
+			oldConn.Close()
 		}
 
 		err = storeConfig(ctx, req.Storage, name, config)
```

Comment:
Are we breaking any assumptions anywhere by calling storeConfig without a lock now?

Reply:
As far as I can tell, no. I don't think there is anything unsafe about multiple calls to `storeConfig`, other than only one of them would win (which is also true even with a lock).